### ISSUE TITLE
Panic occurs when using the custom scheduling scoring plugins

### ISSUE THREAD

**Issue by yxxhero** [kubernetes/kubernetes#106246]
### What happened?

Panic occurs when using the custom scheduling scoring plugins. The customized scheduling scoring plug-in is developed based on the schedule framework. Panic is accidental.

error messsage:
fatal error: concurrent map writes.

### What did you expect to happen?

no panic

### How can we reproduce it (as minimally and precisely as possible)?

None.I think it is caused by the following code:
https://github.com/kubernetes/kubernetes/blob/d3ef2e393b6457b8f4bc407ca01497c51ebe9471/pkg/scheduler/framework/runtime/framework.go#L762-L784

Plugintonodescores will be accessed by multiple goroutines.

### Kubernetes version

<details>
1.22

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider

<details>
None
</details>

### OS version

<details>
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"

Linux localhost.localdomain 5.14.11-1.el7.elrepo.x86_64 #1 SMP Sat Oct 9 09:39:51 EDT 2021 x86_64 x86_64 x86_64 GNU/Linux
```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by 2014BDuck** [comment]
Hi. May I suggest that `pluginToNodeScores` could be accessed by different goroutine, however, `pluginToNodeScores[i]` could be accessed by `i-th` goroutine only.

Would you mind providing more details like the error/fatal output log or the custom plugins codes? So that someone familiar with the scheduler could help to check.

**Comment by ahg-g** [comment]
please provide more info: can you share the exact panic message and the k8s version you are working with?

**Comment by yxxhero** [comment]
@ahg-g k8s v1.18.9. 
no more error message. do you mean goroutine errors? Thanks very much.

**Comment by ahg-g** [comment]
yeah, the panic message itself, it should tell where exactly the panic happened. We did fix a related issue when accessing the cycleState, see https://github.com/kubernetes/kubernetes/issues/95887, so we need to understand if it is the same issue.

v1.18 is out of support, the fix above was applied all the way back to v1.19

**Comment by yxxhero** [comment]
@ahg-g Thanks very much.
